src/modules/c.rs

```diff
@@ -0,0 +1,175 @@
+
+                    let c_compiler = if c_compiler_info.contains("clang") {
+                        "clang"
+                    } else if c_compiler_info.contains("Free Software Foundation") {
+                        "gcc"
+                    } else {
+                        "Unknown compiler"
+                    };
+                    Some(c_compiler).map(Ok)
+                }
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "compiler_version" => {
+                    if config.format.contains("$compiler_version") {
+                        // Clang says ...
+                        //   Apple clang version 13.0.0 ...\n
+                        //   OpenBSD clang version 11.1.0\n...
+                        //   FreeBSD clang version 11.0.1 ...\n
+                        // so seems to have the version always in position 3.
+                        // gcc says ...
+                        //   gcc (OmniOS 151036/9.3.0-il-1) 9.3.0\n...
+                        //   gcc (Debian 10.2.1-6) 10.2.1 ...\n
+                        // so again in position 3.
+                        // But this is CRUFTY AS HELL
+                        VersionFormatter::format_module_version(
+                            module.get_name(),
+                            &c_compiler_info
+                                .as_ref()?
+                                .stdout
+                                .split_whitespace()
+                                .nth(3)?
+                                .to_string(),
+                            config.version_format,
+                        )
+                        .map(Ok)
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `c`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::test::ModuleRenderer;
+    use ansi_term::Color;
+    use std::fs::File;
+    use std::io;
+
+    #[test]
+    fn folder_without_c_files() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+
+        let actual = ModuleRenderer::new("c").path(dir.path()).collect();
+
+        let expected = None;
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn folder_with_c_file() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        File::create(dir.path().join("any.c"))?.sync_all()?;
+
+        // FIXME voodoo here to set up a mock for 'cc --version' claiming to be gcc
+        // NB the test for when it claims to be clang is in the ..._with_h_file test
+        let actual = ModuleRenderer::new("c").path(dir.path()).collect();
```

Comment:
`ModuleRenderer` has `cmd` for dynamic mocking.

Reply:
Usage: https://github.com/starship/starship/blob/master/src/modules/ocaml.rs#L314-L332

Reply:
Ah that looks useful, thanks!